app_test.go

```diff
@@ -84,6 +84,14 @@ func TestNewApp(t *testing.T) {
 	})
 }
 
+type testErrHandler struct {
+	count int
+}
+
+func (eh *testErrHandler) HandleError(err error) {
+	eh.count++
```

Comment:
probably easier to test the error handler by building a func-based implementation.

```
type errorHandlerFunc func(error)

func (f errorHandlerFunc) HandleError(err error) { f(err) }
```

Then in the test where you're looking at the count, you can do,

```
count := 0
h := errHandlerFunc(func(err error) {
  count++
  assert stuff about the error here
})
assert.Equal(t, 1, count)
```

Reply:
Cool, never thought that it can be done like that 👍